File: spring-boot-exception/src/main/java/com/example/springbootexception/UserController.java
package com.example.springbootexception;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
public class UserController {

    public UserController(Hazem user) {
        this.user = user;
    }

    @RequestMapping("/add")
    public String add() {
//        try {
//
//        } catch (Exception e) {
//            model.addAttribute("exception", e.getMessage());
//        }
//        int num = 10 / 0;
        return "add";
    }

    private final Hazem user;

    @RequestMapping("/nullView")
    public String nullView(Model model) {

        System.out.println(user.getHazem());
        return "add";
    }
//
//    @ExceptionHandler(value = {java.lang.ArithmeticException.class})
//    public ModelAndView handlerArithmeticException(Exception e) {
//        ModelAndView modelAndView = new ModelAndView();
//        modelAndView.addObject("exception", e.toString());
//        modelAndView.setViewName("mathError");
//        return modelAndView;
//    }
//
//    @ExceptionHandler(value = {java.lang.NullPointerException.class})
//    public ModelAndView handlerNullPointerException(Exception e) {
//        ModelAndView modelAndView = new ModelAndView();
//        modelAndView.addObject("exception", e.toString());
//        modelAndView.setViewName("nullPointerException");
//        return modelAndView;
//    }

}

File: spring-boot-thymeleaf/src/main/java/com/example/springbootthymeleaf/controller/UserController.kt
package com.example.springbootthymeleaf.controller

import com.example.springbootthymeleaf.models.User
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.RequestMapping
import javax.servlet.http.HttpServletRequest


@Controller
class UserController {

    @RequestMapping("users")
    fun demo2(model: Model): String {
        val lst: MutableList<User> = ArrayList()
        lst.add(User(1, "Tom", 30))
        lst.add(User(2, "Jerry", 29))
        lst.add(User(3, "Nancy", 27))
        model.addAttribute("list", lst)
        return "users_list"
    }

}
